pkg/geoipupdate/database/http_reader.go

```diff
@@ -70,12 +72,29 @@ func NewHTTPReader(
 func (r *HTTPReader) Read(ctx context.Context, editionID, hash string) (*ReadResult, error) {
 	var result *ReadResult
 	var err error
-	err = internal.RetryWithBackoff(
+
+	exp := backoff.NewExponentialBackOff()
+	exp.MaxElapsedTime = r.retryFor
```

Comment:
We have [documented](https://github.com/maxmind/geoipupdate/blob/86a45fb6272809f0b6caa2fa77743516f465734f/CHANGELOG.md?plain=1#L149-L152) that a `RetryFor` value of 0 means that no retries will be performed. I am not sure we would want to change that behavior, especially in a minor release. I didn't think about this when I made the suggestion. I think we could either use [`WithMaxRetries`](https://pkg.go.dev/github.com/cenkalti/backoff/v4#WithMaxRetries) with the value of `0` when `retryFor` is `0` or just modify the function to return a permanent error whenever it is 0.

Reply:
That's a good catch, thank you.